openeo-geotrellis/pom.xml

```diff
@@ -190,7 +190,7 @@
         <dependency>
             <groupId>black.ninia</groupId>
             <artifactId>jep</artifactId>
-            <version>3.9.1</version>
+            <version>4.1.1</version>
```

Comment:
When reading the changelogs for jep I did come across this message for version 4.0.0:
> This is a major version increment and it includes removal of deprecated functionality and several changes making it incompatible with previous versions of Jep. This release also removes support for Python 2.

I will have to double check if there was a specific reason for why version 3.9.1 was chosen, but our use of Jep bindings is minimal so as long as all UDF tests work it is likely that we'll be ok. I will do some extra tests and go over the API changes just in case.

Reply:
Note that the jep version is also defined in these places:
https://github.com/Open-EO/openeo-geopyspark-driver/blob/master/setup.py#L82
https://github.com/Open-EO/openeo-geotrellis-extensions/blob/master/geotrellis-integrationtests/pom.xml#L174

Reply:
I checked the changes for the Jep library between versions 3.9.1 and 4.1.1: 

- The only important change is that setRedirectOutputStreams (whether to redirect Python streams to Java) has been split into redirectStdout, redirectStdErr. (https://github.com/ninia/jep/commit/f67b874b637e24ac3b725f2e89a57b51cded95e1) 
  - This is required to see the UDF logs.
- SharedInterpreters now also have an extra feature but it should not be relevant to our implementation:
[SharedInterpreters now share MemoryManager and java PyObjects. (#421)](https://github.com/ninia/jep/commit/2f6948ac19469b55d4800169a20d21ee418a8658)

I also tested this java build in the geopyspark-driver and it works without issues.